src/db/add_package.rs

```diff
@@ -213,6 +217,30 @@ fn convert_dependencies(pkg: &MetadataPackage) -> Vec<(String, String, String)>
         .collect()
 }
 
+/// Reads features and converts them to Vec<Feature> with default being first
+fn get_features(pkg: &MetadataPackage) -> Vec<Feature> {
+    let mut features = Vec::with_capacity(pkg.features.len());
+    if let Some(subfeatures) = pkg.features.get("default") {
+        features.push(Feature::new("default".into(), subfeatures.clone()));
+    };
+    features.extend(
+        pkg.features
+            .iter()
+            .filter(|(name, _)| name != &"default")
+            .map(|(name, subfeatures)| Feature::new(name.clone(), subfeatures.clone())),
+    );
+    features.extend(get_optional_features(pkg));
+    features
+}
+
+fn get_optional_features(pkg: &MetadataPackage) -> Vec<Feature> {
```

Comment:
This is getting optional *dependencies* - can you put that in the name?
```suggestion
fn get_optional_dependencies(pkg: &MetadataPackage) -> Vec<Feature> {
```

Reply:
Sure, done